src/constraint/nuget.rs

```diff
@@ -0,0 +1,433 @@
+//! # NuGet Versions and their Comparisons:
+//!
+//! FOSSA fetches NuGet vulnerabilities by way of [GHSA](https://github.com/advisories?query=type%3Areviewed+ecosystem%3Anuget).
+//!
+//! ## NuGet Versioning and SemVer 2
+//!
+//! NuGet versions are generally compatible with SemVer 2.
+//! They document their differences [here](https://learn.microsoft.com/en-us/nuget/concepts/package-versioning?tabs=semver20sort#where-nugetversion-diverges-from-semantic-versioning):
+//!
+//! In brief, the known deviations from SemVer 2 are:
+//! - NuGetVersion supports a 4th version segment, Revision. So they are `Major.Minor.Patch.Revision`.
+//! - NuGetVersion do not consider build metadata.
+//! - NuGetVersion only requires the major segment to be defined. All others are optional, and are equivalent to zero. This means that 1, 1.0, 1.0.0, and 1.0.0.0 are all accepted and equal.
+//! - NuGetVersion uses case insensitive string comparisons for pre-release components. This means that 1.0.0-alpha and 1.0.0-Alpha are equal.
```

Comment:
This is true but Nuget also supports Semver 1.0.0, and there are specific rules for comparing them if they diverge: https://github.com/fossas/sparkle/blob/76e7e29ba9faae5b439cd80c6b82cd94f9d843de/lib/fetcher/src/client/nuget.rs#L490-L594; see also https://learn.microsoft.com/en-us/nuget/concepts/package-versioning?tabs=semver20sort#pre-release-versions.

Please add support for semver 1 comparison as well. It may be useful to compare with the existing version comparison tests for the nuget fetcher and/or steal from its implementation: https://github.com/fossas/sparkle/blob/76e7e29ba9faae5b439cd80c6b82cd94f9d843de/lib/fetcher/tests/it/client/nuget/version.rs#L1